tests/integration/utils.py

```diff
@@ -0,0 +1,238 @@
+
+        return openapi_type
+
+    # Helper function to check resource object schema
+    def __check_resource_schema(resource):
+        # Check resource type
+        self.assertEqual(resource['type'], schema['type']['enum'][0])
+        # Check resource attributes
+        actual_attributes = resource['attributes']
+        expected_attributes = __get_schema_attributes()
+        __check_attributes_schema(actual_attributes, expected_attributes)
+
+    # Helper function to check error object schema
+    def __check_error_schema(error):
+        # Check error attributes
+        actual_attributes = error
+        expected_attributes = schema
+        __check_attributes_schema(actual_attributes, expected_attributes)
+
+    # Helper function to check through all attributes
+    def __check_attributes_schema(actual_attributes, expected_attributes):
+        for field, actual_value in actual_attributes.items():
+            expected_attribute = expected_attributes[field]
+            expected_type = __get_attribute_type(expected_attribute)
+            if (actual_value):
+                self.assertIsInstance(actual_value, expected_type)
+
+    status_code = response.status_code
+    content = get_json_content(self, response)
+
+    # TODO: Add self-link testing
+    # Basic tests for successful/error response
+    try:
+        if status_code == 200:
+            resource_data = content['data']
+            if isinstance(resource_data, list):
+                for resource in resource_data:
+                    __check_resource_schema(resource)
+            else:
+                __check_resource_schema(resource_data)
+        elif status_code >= 400:
+            errors_data = content['errors']
+            self.assertIsInstance(errors_data, list)
+            for error in errors_data:
+                __check_error_schema(error)
+    except KeyError as error:
+        self.fail(error)
+
+
+# Check response of an endpoint against path cases for schema, data validity
+def test_path_request(self, endpoint, resource, response_code, test_cases,
+                      test_assertion=None):
+    for test_case in test_cases:
+        test_request(self, endpoint + "/" + test_case, resource, response_code,
+                     "", "", test_assertion)
+
+
+# Check response of an endpoint against query cases for schema, data validity
+def test_request(self, endpoint, resource, response_code, param, test_cases,
+                 test_assertion=None):
+
+    schema = get_resource_schema(self, resource)
+    for test_case in test_cases:
+        response = make_request(self, endpoint, response_code,
+                                params={param: test_case})
+        check_schema(self, response, schema)
+
+        if (test_assertion):
+            for resource in response.json()['data']:
+                actual_case = resource['attributes'][param]
+                test_assertion(self, actual_case, test_case)
+
+
+class assertion_tests:
```

Comment:
The methods in this class use `assertTrue()` which comes from the `unittest.TestCase` class. I think this class will need to extend that class for clarification. The methods would fail if `self` wasn't an instance of `unittest.TestCase`.